benchmarks/benchmarks/common.py

```diff
@@ -25,40 +25,90 @@
     'complex256',
 ]
 
+
+def memoize(func):
+    result = []
+    def wrapper(*a, **kw):
+        if not result:
+            result.append(func(*a, **kw))
+        return result[0]
+    return wrapper
```

Comment:
That is super cool!
Obligatory demand for a pony: is there a simple way to compare "now" versus
"last release" to check for regressions? After all the fun we had with
1.10, automated release qa has been on my mind...
On Dec 17, 2015 9:57 AM, "Pauli Virtanen" <notifications@github.com> wrote:

> In benchmarks/benchmarks/common.py
> <https://github.com/numpy/numpy/pull/6829#discussion_r47937917>:
>
> > @@ -25,40 +25,90 @@
> >      'complex256',
> >  ]
> >
> > +
> > +def memoize(func):
> > +    result = []
> > +    def wrapper(*a, **kw):
> > +        if not result:
> > +            result.append(func(*a, **kw))
> > +        return result[0]
> > +    return wrapper
>
> FWIW: https://pv.github.io/numpy-bench/ (updates daily)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/6829/files#r47937917>.
>